pallets/pallet-bonded-coins/src/lib.rs

```diff
@@ -128,15 +131,7 @@ pub mod pallet {
 		type RuntimeHoldReason: From<HoldReason>;
 
 		/// The type used for the curve parameters.
-		type CurveParameterType: Parameter
-			+ Member
-			+ FixedPointNumber<Inner = u128>
-			+ TypeInfo
-			+ MaxEncodedLen
-			+ SquareRoot
-			+ Zero
-			+ CheckedAdd
-			+ CheckedSub;
+		type CurveParameterType: Parameter + Member + FixedSigned + MaxEncodedLen + PartialOrd<I9F23> + From<I9F23>;
```

Comment:
Maybe introduce another type for the creation step. Only Unsigned parameters should be allowed.